### PR TITLE
Notifications preview command

### PR DESCRIPTION
**Pull request by kpuputti**
This PR adds a new `notifications preview` command. It takes a `--template` options that points to a template directory and:

1. Sends the template subject and html to the `/notifications/preview` API endpoint
2. Prints out the rendered preview subject and text
3. Creates a temporary HTML file with the subject injected in the `<title>`
4. Opens the created HTML file in a browser

## Template with errors

Screen Shot 2019-11-05 at 14 03 05

## CLI output of preview subject and text

Screen Shot 2019-11-05 at 14 07 55

## Preview HTML opened automatically in a browser

See the subject injected in the `<title>`

Screen Shot 2019-11-05 at 14 02 54

### CODE REVIEW

File: src/sharetribe/flex_cli/commands/notifications/preview.cljs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          :handler #'preview
+          :desc "render a preview of an email template"
+          :opts [{:id :template
+                  :long-opt "--template"
+                  :desc "path to a template directory"
+                  :required "TEMPLATE_DIR"
+                  :missing "--template is required"}]})
+
+(defn bold [str]
+  (.bold chalk str))
```
Comment (ovan) [inline src/sharetribe/flex_cli/commands/notifications/preview.cljs]: I am growing increasingly uncomfortable of the styling fns being here and there and every command defining individually how to style the output. Mikko started something with sharetribe.flex-cli.view, then we have some output formatting in io-util and a lot of it in command fns. We can table this discussion until we get to the story of "unify output" but I'd rather not delay that work indefinitely but e.g. work with that when we have the main scope together and/or as a cool down task after this.

File: src/sharetribe/flex_cli/process_util.cljs [hunk 2]
```diff
@@ -52,6 +56,7 @@
 (defmethod exception/format-exception :process.util/new-process-api-call-failed [_ _ data]
```
Comment (ovan) [inline src/sharetribe/flex_cli/process_util.cljs]: Ok, now I see the answer to my question about why this was new-process-api-call-failed. That said, this keyword is quite misleading and feels like something is not fully thought out but rather grew a bit out of control. Any idea how to make the error throwing from failed api calls and handling it more sensical in terms of naming / structuring?

File: src/sharetribe/flex_cli/commands/notifications/preview.cljs [hunk 3]
```diff
@@ -0,0 +1,60 @@
+(ns sharetribe.flex-cli.commands.notifications.preview
+  (:require [clojure.string :as str]
+            [clojure.core.async :as async :refer [go <!]]
+            [sharetribe.flex-cli.api.client :refer [do-post]]
+            [sharetribe.flex-cli.async-util :refer [<? go-try]]
+            [sharetribe.flex-cli.io-util :as io-util]
+            [sharetribe.flex-cli.api.client :as api.client]
+            [chalk]
+            ["tmp" :rename {fileSync tmp-file-sync}]
+            ["open" :as open]))
```
[the remark this answers is not in the document]
Comment (kpuputti) [reply src/sharetribe/flex_cli/commands/notifications/preview.cljs]: `[open]` seems to work fine. I took example from an earlier import where renaming was required. And that was taken from some reference where JS import syntaxes were mapped to respective CLJS imports. But it seems that the quotation marks are not needed.

File: src/sharetribe/flex_cli/process_util.cljs (hunk 2 above)
Comment (kpuputti) [reply src/sharetribe/flex_cli/process_util.cljs]: Yeah, this has definitely grown out of its original "bounds". I started reorganizing these errors.

**Pull request merged**